src/test/ui/pub/pub-restricted-error-fn.stderr

```diff
@@ -6,11 +6,25 @@ LL | pub(crate) () fn foo() {}
    |
    = help: you likely meant to define an item, e.g., `pub(crate) fn foo() {}`
 
-error: expected item, found `(`
+error: statements cannot reside in modules
   --> $DIR/pub-restricted-error-fn.rs:1:12
    |
 LL | pub(crate) () fn foo() {}
-   |            ^ expected item
+   |            ^^
```

Comment:
This seems less than ideal. I think when we're midway some item head, we should not emit the suggestion.

Reply:
But is this a common case? And if it is, does the first error not suggest what the problem is and how to fix it? And is it not better to recover than to fatally error? Otherwise we could have some flag when calling `parse_item_common_` which signifies that we've parsed at least the visibility / defaultness, if you would prefer the error as-is, though that is adding code complexity, so consider whether this is common enough to be worth it.

Reply:
I read your comment again; it makes more sense now since you're talking only about the *suggestion* -- I think we could use the same flag and pass it in to avoid everything except for the primary message, and instead do "remove the statements".